pkg/builtin/multi/round.go

```diff
@@ -49,23 +82,302 @@ func init() {
 	}
 	overload.OpTypes[builtin.Round] = overload.Multi
 	overload.MultiOps[builtin.Round] = []*overload.MultiOp{
+		{
+			Min:        1,
+			Max:        2,
+			Typ:        types.T_uint8,
+			ReturnType: types.T_uint8,
+			Fn: func(vecs []*vector.Vector, proc *process.Process, cs []bool) (*vector.Vector, error) {
+				digits := int64(0)
+				vs := vecs[0].Col.([]uint8)
+				if len(vecs) > 1 {
+					if !cs[1] && vecs[1].Typ.Oid != types.T_int64 {
+						return nil, errors.New("The second argument of the round function must be an int64 constant")
+					}
+					digits = vecs[1].Col.([]int64)[0]
+				}
+				if vecs[0].Ref == 1 || vecs[0].Ref == 0 {
+					vecs[0].Ref = 0
+					round.RoundUint8(vs, vs, digits)
+					return vecs[0], nil
+				}
+				vec, err := process.Get(proc, int64(len(vs)), types.Type{Oid: types.T_uint8, Size: 1})
+				if err != nil {
+					return nil, err
+				}
+				rs := encoding.DecodeUint8Slice(vec.Data)
+				rs = rs[:len(vs)]
+				vec.Col = rs
```

Comment:
I think there need some code here, like

`for I := 1; I < len(vecs); i++ {
if !cs[I] && vecs[I] <= 1 {
       process.Put(proc, vecs[I]);
  }
}`
and other functions also.

Reply:
the second argument of round function should be a int64 constant otherwise an error will be thrown